File: Part 2 - Regression/Section 5 - Multiple Linear Regression/Multiple_LR.R
#Importing Dataset
dataset = read.csv('50_Startups.csv')

#Encoding Categorical values
dataset$State = factor(dataset$State,
                       levels = c('New York','California','Florida'),
                       labels = c(1,2,3))

#Spliting dataset to training sets and testing sets
library(caTools)
set.seed(123)
split = sample.split(dataset$Profit,SplitRatio=0.8)
training_set = subset(dataset,split == TRUE)
test_set = subset(dataset,split == FALSE)

#Fitting MLR to Training Set
regressor = lm(formula = Profit ~ .,data = training_set)

#Predicting the Test set results
y_pred = predict(regressor, newdata = test_set)

#Building the optimal model using Backward elimination
regressor_opt = lm(formula = Profit ~ R.D.Spend + Administration + Marketing.Spend + State,data = dataset)
summary(regressor_opt)

regressor_opt = lm(formula = Profit ~ R.D.Spend + Administration + Marketing.Spend ,data = dataset)
summary(regressor_opt)

regressor_opt = lm(formula = Profit ~ R.D.Spend  + Marketing.Spend ,data = dataset)
summary(regressor_opt)

regressor_opt = lm(formula = Profit ~ R.D.Spend ,data = dataset)
summary(regressor_opt)

File: Part 2 - Regression/Section 5 - Multiple Linear Regression/Multiple_LR.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun Mar  8 17:18:44 2020

@author: macbook
"""

#Importing Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#Importing Dataset
dataset = pd.read_csv('50_Startups.csv')
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,4].values

#Encoding Categorical Data
#from sklearn.preprocessing import LabelEncoder,OneHotEncoder
#labelencoder_X = LabelEncoder()
#X[:,3] = labelencoder_X.fit_transform(X[:,3])
#ohe = OneHotEncoder(categorical_features = [3])
#X = ohe.fit_transform(X).toarray()

from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
ct = ColumnTransformer([('encoder', OneHotEncoder(categories='auto'), [3])], remainder='passthrough')
X = np.array(ct.fit_transform(X), dtype=np.float)

#Avoiding the  dummy variable trap
X = X[:,1:]

#Splitting the dataset into training set and testing set
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size = 0.2,random_state = 0)

#Fitting MLR to Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#Predicting the Test set results
y_pred = regressor.predict(X_test)

#Building the optimal model using Backward elimination
import statsmodels.api as sm
X = np.append(arr = np.ones((50,1)).astype(int) , values = X ,axis =1 )
X_opt = X[:,[0,1,2,3,4,5]]
regressor_OLS = sm.OLS(endog = y,exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:,[0,1,3,4,5]]
regressor_OLS = sm.OLS(endog = y,exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:,[0,3,4,5]]
regressor_OLS = sm.OLS(endog = y,exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:,[0,3,5]]
regressor_OLS = sm.OLS(endog = y,exog = X_opt).fit()
regressor_OLS.summary()

X_opt = X[:,[0,3]]
regressor_OLS = sm.OLS(endog = y,exog = X_opt).fit()
regressor_OLS.summary()







